pyccel/stdlib/lists/lists.c

```diff
@@ -0,0 +1,212 @@
+#include "lists.h"
+
+
+t_list   *allocate_list(size_t size, t_type type, void *elemnts)
+{
+    t_list *list;
+    size_t tsize = tSizes[type];
```

Comment:
@azouiten There's a slight problem regarding lists initialized with no elements, we have no specific type for them. I suggest to add `lst_unkown` to `enum e_type`, which could default to 1 byte in `tSizes` for example, and then reassign a new type to the list structure depending on the firsts elements type. This is involves the functions `append`, `pop` and `extend`.

Reply:
I don't think this is a good solution as there will be no way of implementing it in Fortran. I'm not sure what the work around should be though. Perhaps we could do this on the pyccel side so that by the time we start to print the type is known?